### PR TITLE
39 add model organization

### LINKED ISSUE #39: Add model: Team
For now, a Team has many users and a User has many Teams.

### PR DESCRIPTION
**Pull request by eugenk**
This shall fix #39. This is based on #37. The first new commit is "Add model: Organization. ab4f06c".

Todo
* [x] Rebase on master as soon as #37 is merged.

### CODE REVIEW

File: spec/factories/teams_factory.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# frozen_string_literal: true
```
Comment (ebolloff) [inline spec/factories/teams_factory.rb]: Can you rename the file to `organizations_factory.rb`?

File: app/models/user.rb [hunk 2]
```diff
@@ -6,9 +6,6 @@ class User < OrganizationalUnit
```
Comment (ebolloff) [inline app/models/user.rb]: Could you add a method to get all repositories a user has access to via all his organizations.

File: spec/factories/teams_factory.rb (hunk 1 above)
Comment (eugenk) [inline spec/factories/teams_factory.rb]: Done in a commit --amend.

**Comment by eugenk** [comment]
The last three commits add the association in different styles. 

1. b0cf4f89e727e742d6f94ba86461ccf156457674 adds a method, which is very easy to read, but performs slowly.
1. 4fe468623829fe8b0baa1ba3fa82131b83617af4 replaces it by an association that uses a subquery. It should be faster than the previous method and is still quite easy to read, but it creates a `WHERE id IN (...)` clause, which is usually slow.
1. 83976cf495a7b3b26a1e280559e8baa7917aed27 replaces it by an association that uses a join query. It is harder to read but should perform the best.

I have put it in three different commits to make it easier to understand and to have tests that confirm that the results of the different methods are all equal.

**Pull request merged**